autotest/gcore/hfa_rfc40.py

```diff
@@ -190,15 +190,15 @@ def ReadAndCheckValues(fname, numrows):
         raise HFATestError("double column does not match")
 
     data = rat.ReadAsArray(2, 0, 10)
-    if not (data == STRING_DATA.astype(numpy.character)).all():
```

Comment:
The deprecation warning is:
```Converting `np.character` to a dtype is deprecated. The current result is `np.dtype(np.str_)` which is not strictly correct. Note that `np.character` is generally deprecated and 'S1' should be used.```

Presume there's no practical difference in this case between `'S'` and `'S1'`? Or IIUC `str`?

Reply:
`'S1'` will only produce a single character, which doesn't work for the variable length strings here. And `str` is `'U'`. Alternatively to `'S'`, this could use `bytes`.